disco_aws_automation/disco_rds.py

```diff
@@ -170,80 +381,142 @@ def update_cluster(self, database_name):
+        clone_db_identifier = RDS.get_instance_identifier(self.vpc_name, source_db)
 
-    def setup_dns(self, instance_identifier):
-        """
-        Setup Domain Name Lookup using Route 53
-        """
-        start_time = time.time()
-        instance_endpoint = keep_trying(RDS_STARTUP_TIMEOUT, self._get_instance_address, instance_identifier)
-        logging.info("Waited %s seconds for RDS to get an address", time.time() - start_time)
-        disco_route53 = DiscoRoute53()
-        instance_record_name = '{0}.{1}.'.format(instance_identifier, self.domain_name)
+        instance_params = self.get_instance_parameters(source_vpc, source_db)
 
-        # Delete and recreate DNS record for this Instance
-        disco_route53.delete_record(self.domain_name, instance_record_name, 'CNAME')
-        disco_route53.create_record(self.domain_name, instance_record_name, 'CNAME', instance_endpoint)
+        if self._get_db_instance(clone_db_identifier):
+            raise RDSEnvironmentError(
+                'Cannot create clone instance {0} because a database already exists with that name'
+                .format(clone_db_identifier)
+            )
 
-    def spinup_alarms(self, database_name):
+        # override some parameters to use the new instance id
+        instance_params['DBInstanceIdentifier'] = clone_db_identifier
+        instance_params['DBSubnetGroupName'] = clone_db_identifier
+        instance_params['DBParameterGroupName'] = clone_db_identifier
+
+        self.recreate_db_subnet_group(instance_params["DBSubnetGroupName"])
+
+        group_name = instance_params["DBParameterGroupName"]
+        group_family = RDS.get_db_parameter_group_family(
+            instance_params["Engine"], instance_params["EngineVersion"])
+        logging.debug("creating parameter group %s with family %s", group_name, group_family)
+
+        # create a parameter group using the parameters of the source db
+        self.recreate_db_parameter_group(source_vpc, source_db, group_name, group_family)
+
+        self.create_db_instance(instance_params,
+                                custom_snapshot=self.get_final_snapshot(source_db_identifier))
+
+        # Create/Update CloudWatch Alarms for this instance
+        self.spinup_alarms(source_db)
+
+        # Create a DNS record for this instance
+        self.setup_dns(clone_db_identifier)
+
+    def run(self):
+        self.update_cluster()
+
+
+class DiscoRDS(object):
+    """Class for doing RDS operations on a given environment"""
+
+    def __init__(self, vpc):
+        """Initialize class"""
+        self.config_aws = read_config()
+        self.config_rds = read_config(config_file=DEFAULT_CONFIG_FILE_RDS)
+        self.client = boto3.client('rds')
+        self.vpc = vpc
+        self.disco_vpc_sg_rules = DiscoVPCSecurityGroupRules(vpc, vpc.boto3_ec2)
+        self.vpc_name = vpc.environment_name
+        if self.vpc_name not in ['staging', 'production']:
```

Comment:
This seems unnecessarily hard-coded. Might be better to copy-paste the config logic for properly handling the `param@env` syntax than to hard code VPC names that are not necessarily universal.

Another point in favor of the eventual `DiscoConfig` class.